src/Handler/StreamHandler.php

```diff
@@ -517,6 +517,8 @@ private function add_progress(RequestInterface $request, array &$options, $value
             $params,
             static function ($code, $a, $b, $c, $transferred, $total) use ($value) {
                 if ($code == \STREAM_NOTIFY_PROGRESS) {
+                    assert(\is_callable($value));
```

Comment:
This could potentially be fixed by a `callable` docblock, but given the assumption that `$value` is only used as such when `$code` is `\STREAM_NOTIFY_PROGRESS`, it will have to be a union and need either an assertion or a `@var` tag override

Reply:
again, please don't use the assert function.